DigitalLearningSolutions.Data.Tests/DataServices/UserDataServiceTests.cs

```diff
@@ -318,7 +319,7 @@ public void RemoveDelegateUser_cannot_remove_delegate_user_with_started_session(
                     Action action = () => userDataService.RemoveDelegateUser(id);
 
                     // Then
-                    action.Should().Throw<UserAccountInvalidStateException>();
+                    action.Should().Throw<SqlException>();

```

Comment:
Nice - does this give a reasonable error in the logs so it's clear where/why the error came from?

Reply:
It does, the SQL error identifies the foreign key constraint that was violated.